Make Inductor cpp backend enable_floating_point_contract_flag to take string

hl475 opened the pull request:
Differential Revision: D66269001




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

hl475 commented:
@pytorchbot label "topic: not user facing"

desertfire commented on torch/_inductor/config.py:
```diff
@@ -854,6 +854,14 @@ class cpp:
     )
 
+    # Use ffp-contract=fast when compiling
+    enable_floating_point_contract_fast_flag = (
```
Makes sense in general, but can we combine with the existing `enable_floating_point_contract_flag`? Adding another one can be confusing.

desertfire commented on torch/_inductor/cpp_builder.py:
```diff
@@ -538,6 +538,9 @@ def _get_optimization_cflags(cpp_compiler: str) -> List[str]:
             cflags.append("fno-unsafe-math-optimizations")
         if not config.cpp.enable_floating_point_contract_flag:
             cflags.append("ffp-contract=off")
+        if config.cpp.enable_floating_point_contract_fast_flag:
```
Similar to my last comment,  I think now we can have both "ffp-contract=off" and "ffp-contract=fast" in the compile flags.

hl475 committed:
Make Inductor cpp backend enable_floating_point_contract_flag to take string (#143450)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/143450

Test Plan: Sandcastle

Reviewed By: desertfire

Differential Revision: D66269001
